### ISSUE TITLE
Crash when evaluating query that uses @count on list of primitive

### ISSUE THREAD

**Issue by mitevdev** [realm/realm-cocoa#5491]
```
class Entry: Object {
   let tags = List<String>()
}
```
and then later:

`realm.objects(Entry.self).filter("tags.@count == 0")`

## Goals
fetch entries having no tags

## Expected Results
Results<Entry> instance 

## Actual Results
Exception EXC_BAD_ACCESS (code=1, address=0x0) in 

> * thread #10, name = 'RLMRealm notification listener', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
>   * frame #0: 0x0000000100462794 Realm`realm::LinkMap::map_links(unsigned long, unsigned long, realm::LinkMapFunction&) + 52
>     frame #1: 0x000000010046242b Realm`realm::LinkCount::evaluate(unsigned long, realm::ValueBase&) + 65
>     frame #2: 0x000000010045ff93 Realm`realm::Compare<realm::Equal, long long, realm::Subexpr, realm::Subexpr>::find_first(unsigned long, unsigned long) const + 113
>     frame #3: 0x00000001006ed410 Realm`realm::ParentNode::aggregate_local(realm::QueryStateBase*, unsigned long, unsigned long, unsigned long, realm::SequentialGetterBase*) + 144
>     frame #4: 0x000000010067c15a Realm`realm::Query::aggregate_internal(realm::Action, realm::DataType, bool, realm::ParentNode*, realm::QueryStateBase*, unsigned long, unsigned long, realm::SequentialGetterBase*) const + 314
>     frame #5: 0x0000000100680ad8 Realm`realm::Query::find_all(realm::TableViewBase&, unsigned long, unsigned long, unsigned long) const + 440
>     frame #6: 0x0000000100681ff6 Realm`realm::Query::find_all(unsigned long, unsigned long, unsigned long) + 102
>     frame #7: 0x000000010041485b Realm`realm::_impl::ResultsNotifier::run() + 119
>     frame #8: 0x00000001004054c2 Realm`realm::_impl::RealmCoordinator::run_async_notifiers() + 1730
>     frame #9: 0x0000000100404da8 Realm`realm::_impl::RealmCoordinator::on_change() + 18
>     frame #10: 0x00000001003eba11 Realm`realm::_impl::ExternalCommitHelper::listen() + 191
>     frame #11: 0x00000001003ebe2e Realm`std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0>::operator()() + 18
>     frame #12: 0x00000001003ebdba Realm`std::__1::__async_assoc_state<void, std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0> >::__execute() + 26
>     frame #13: 0x00000001003ebf4b Realm`std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (std::__1::__async_assoc_state<void, std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0> >::*)(), std::__1::__async_assoc_state<void, std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0> >*> >(void*, void*) + 62
>     frame #14: 0x00000001052f26c1 libsystem_pthread.dylib`_pthread_body + 340
>     frame #15: 0x00000001052f256d libsystem_pthread.dylib`_pthread_start + 377
>     frame #16: 0x00000001052f1c5d libsystem_pthread.dylib`thread_start + 13

## Steps to Reproduce
create and store Instance of Entry and try to filter using @count aggregate function == 0

## Version of Realm and Tooling
```
ProductName:	Mac OS X
ProductVersion:	10.13.1
BuildVersion:	17B48

/Applications/Xcode.app/Contents/Developer
Xcode 9.1
Build version 9B55


(not in use here)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)


(not in use here)

/usr/bin/git
git version 2.13.6 (Apple Git-96)
```

**Comment by bdash** [comment]
Note that queries involving arrays of primitives are not yet supported (https://github.com/realm/realm-cocoa/issues/5334). We should be catching this and throwing a meaningful exception, though.

**Comment by Bobozee** [comment]
Correct. This is exactly what I was expecting once I realised that arrays of primitives are not supported in queries yet.

**Comment by tgoyne** [comment]
This was implemented in 10.7.0.
